source/adapters/hip/CMakeLists.txt

```
@@ -10,13 +10,13 @@ set(UR_HIP_PLATFORM "AMD" CACHE STRING "UR HIP platform, AMD or NVIDIA")
 
 # Set default ROCm installation directory
 set(UR_HIP_ROCM_DIR "/opt/rocm" CACHE STRING "ROCm installation dir")
-
-set(UR_HIP_INCLUDE_DIR "${UR_HIP_ROCM_DIR}/include")
-
-set(UR_HIP_HSA_INCLUDE_DIR "${UR_HIP_ROCM_DIR}/hsa/include")
-
-# Set HIP lib dir
-set(UR_HIP_LIB_DIR "${UR_HIP_ROCM_DIR}/lib")
+# Allow custom location of HIP/HSA include and HIP library directories
+set(UR_HIP_INCLUDE_DIR "${UR_HIP_ROCM_DIR}/include" CACHE FILEPATH
+  "Custom ROCm HIP include dir" FORCE)
+set(UR_HIP_HSA_INCLUDE_DIR "${UR_HIP_ROCM_DIR}/hsa/include" CACHE FILEPATH
+  "Custom ROCm HSA include dir" FORCE)
+set(UR_HIP_LIB_DIR "${UR_HIP_ROCM_DIR}/lib" CACHE FILEPATH
+  "Custom ROCm HIP library dir" FORCE)
```

Having `FORCE` here negates all the supposed configurability. Even if user sets, for example, `-DUR_HIP_HSA_INCLUDE_DIR=/opt/rocm/include/hsa/`, it is still overridden to `${UR_HIP_ROCM_DIR}/hsa/include`.


```suggestion
set(UR_HIP_INCLUDE_DIR "${UR_HIP_ROCM_DIR}/include" CACHE FILEPATH
  "Custom ROCm HIP include dir")
set(UR_HIP_HSA_INCLUDE_DIR "${UR_HIP_ROCM_DIR}/hsa/include" CACHE FILEPATH
  "Custom ROCm HSA include dir")
set(UR_HIP_LIB_DIR "${UR_HIP_ROCM_DIR}/lib" CACHE FILEPATH
  "Custom ROCm HIP library dir")
```

---

Oh, one more nitpick here (this one does not matter much): for directories, it should be `PATH`, not `FILEPATH`: https://cmake.org/cmake/help/v3.28/command/set.html#set-cache-entry

> `FILEPATH`
>
>    Path to a file on disk. [cmake-gui(1)](https://cmake.org/cmake/help/v3.28/manual/cmake-gui.1.html#manual:cmake-gui(1)) offers a file dialog.
>
> `PATH`
>
>    Path to a directory on disk. [cmake-gui(1)](https://cmake.org/cmake/help/v3.28/manual/cmake-gui.1.html#manual:cmake-gui(1)) offers a file dialog.